moto/ec2/models/amis.py

```diff
@@ -276,10 +276,22 @@ def describe_images(
             if exec_users:
                 tmp_images = []
                 for ami in images:
-                    for user_id in exec_users:
-                        for lp in ami.launch_permissions:
-                            if lp.get("UserId") == user_id:
-                                tmp_images.append(ami)
+                    if ami.is_public:
+                        tmp_images.append(ami)
+                    else:
+                        # support filtering by ExecutableUsers=['self']
+                        if "self" in exec_users:
+                            exec_users = list(
+                                map(
+                                    lambda o: self.account_id if o == "self" else o,  # type: ignore[attr-defined]
+                                    exec_users,
+                                )
+                            )
```

Comment:
```suggestion
                            exec_users = list(set(exec_users))
                            exec_users.remove("self")
                            exec_users.append(self.account_id)
```
If I understand the logic correctly, this should be equivalent? And better readable IMO.
The `set` operation is just there for the corner-case that users send a request like `exec_users=["self", "self"]`.